packages/core/examples/registration/index.ts

```diff
@@ -0,0 +1,852 @@
+
+    logStatus(
+      `transformParameterObject:\n${stringify(transformParameterObject, 4)}`,
+      true
+    );
+
+    logStatus(`Total time: ${(totalTime / 1000).toFixed(3)} seconds`);
+    logStatus('Registration complete');
+  },
+});
+
+// =============================================================================
+
+async function getElastixParameterMap(
+  transformName: string
+): Promise<DefaultParameterMapResult> {
+  const parameterMapOptions: DefaultParameterMapOptions = {
+    numberOfResolutions: defaultNumberOfResolutions,
+    finalGridSpacing: defaultFinalGridSpacing,
+  };
+
+  const parameterMap: DefaultParameterMapResult = await defaultParameterMap(
+    webWorker,
+    transformName,
+    parameterMapOptions
+  );
+
+  webWorker = parameterMap.webWorker;
+
+  return parameterMap;
+}
+
+async function loadAndCacheAllParameterMaps() {
+  for (let i = 0, len = transformNames.length; i < len; i++) {
+    const transformName = transformNames[i];
+    const { parameterMap } = await getElastixParameterMap(transformName);
+
+    defaultParameterMaps[transformName] = parameterMap;
+    console.log(`Default parameter map (${transformName}):`, parameterMap);
+  }
+}
+
+/**
+ * Update the screen with all parameter values for a given transformation
+ * @param transformName - Transformation name
+ */
+function loadParameterMap(transformName: string) {
+  const parameterMap = defaultParameterMaps[transformName];
+
+  // Update the current parameter map that shall be updated on every input change
+  currentParameterMap = parameterMap;
+
+  // For each parameters that has settings which means an input field on the screen
+  Object.keys(parametersSettings).forEach((parameterName) => {
+    const parameterSettings = parametersSettings[parameterName];
+    const parameterValues = parameterMap[parameterName];
+    const input = document.getElementById(parameterName) as HTMLInputElement;
+
+    // Disable the input field if there is the parameter is not
+    // in the default parameter map
+    input.disabled = !parameterValues;
+
+    if (!parameterValues) {
+      return;
+    }
+
+    // Get the first value because the values are always stored as an array
+    const parameterValue = parameterValues[0];
+
+    // Update the input field
+    input.value = parameterValue;
+
+    // Some parameters needs to update a global variable
+    parameterSettings.onLoad?.(parameterValue);
+  });
+}
+
+/**
+ * Get the current date/time ("YYYY-MM-DD hh:mm:ss.SSS")
+ */
+function getFormatedDateTime() {
+  const now = new Date();
+  const day = `0${now.getDate()}`.slice(-2);
+  const month = `0${now.getMonth() + 1}`.slice(-2);
+  const year = now.getFullYear();
+  const hours = `0${now.getHours()}`.slice(-2);
+  const minutes = `0${now.getMinutes()}`.slice(-2);
+  const seconds = `0${now.getSeconds()}`.slice(-2);
+  const ms = `00${now.getMilliseconds()}`.slice(-3);
+
+  return `${year}-${month}-${day} ${hours}:${minutes}:${seconds}.${ms}`;
+}
+
```

Comment:
this can be in a utils file that sits beside this example, just to save space on this example

Reply:
along with other utilities